src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AllocationResourceIntTest.kt

```diff
@@ -873,6 +879,49 @@ class AllocationResourceIntTest : IntegrationTestBase() {
             PayBandTestParameter("5", yesterday, today),
             PayBandTestParameter("7", tomorrow, null),
           ),
+          // Handling missing pay bands
+          Arguments.of(
+            "A missing pay band continues to be missing",
+            listOf<PayBandTestParameter>(),
+            null,
+            null,
+            null,
+          ),
```

Comment:
A nuch of new tests to handle missing pay bands. The arguments are description, existing pay bands, requested pay band, and finally the first 2 pay bands to assert on after the update.